code/game/objects/items/cigs_lighters.dm

```diff
@@ -157,12 +157,16 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	. = ..()
 
 /obj/item/clothing/mask/cigarette/attackby(obj/item/W, mob/user, params)
-	if(!lit && smoketime > 0)
-		var/lighting_text = W.ignition_effect(src, user)
-		if(lighting_text)
-			light(lighting_text)
-	else
+	if(lit || smoketime <= 0)
 		return ..()
+	if(!reagents.has_reagent(/datum/reagent/oxygen)) //cigarettes need oxygen
+		var/datum/gas_mixture/air = return_air()
+		if(!air.has_gas(/datum/gas/oxygen, 1)) //or oxygen on a tile to burn
```

Comment:
It's rare, but return_air can in some cases return a null value. You should check for it here.

Reply:
That is some strange behavior...  I wonder if it's caused by the tile z-level being completely removed.  (like if a bomb goes off on ice and you can fall through the floor, etc.)

Regardless, it has been addressed with d85e3b4